pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -417,12 +417,16 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 
 		action.Timeout = d
 		if util.IsIstioVersionGE18(node) {
-			if maxDuration := action.MaxStreamDuration; maxDuration != nil {
-				maxDuration.MaxStreamDuration = d
-			} else {
-				action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{
-					MaxStreamDuration: d,
-				}
+			if action.MaxStreamDuration == nil {
+				action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{}
+			}
+			action.MaxStreamDuration.MaxStreamDuration = d
+
+			// Set the GrpcTimeoutHeaderMax so that Envoy respects grpc-timeout header.
+			// Only set if explicit timeout is defined otherwise Envoy will just use grpc-timeout header
+			// instead of disabling the timeout which is Istio's default behavior.
+			if d.Seconds > 0 {
```

Comment:
curious, why should this be greater than 1 second

Reply:
If we do not have that check, i.e if it is set to zero, Envoy will use what ever is the `grpc-timeout` header so it will not honour the `ISTIO_DEFAULT_REQUEST_TIMEOUT`  which is zero.

Reply:
Could i set `ISTIO_DEFAULT_REQUEST_TIMEOUT = 100ms`

Reply:
Yes. I missed that it is a duration var. Fixed it now. PTAL.